src/SplitFlapMqtt.cpp

```diff
@@ -0,0 +1,108 @@
+    String mdns = settings.getString("mdns");
+    String name = settings.getString("name");
+
+    topic_command = "splitflap/" + mdns + "/set";
+    topic_state   = "splitflap/" + mdns + "/state";
+    topic_avail   = "splitflap/" + mdns + "/availability";
+    topic_config_text  = "homeassistant/text/splitflap_text_" + mdns + "/config";
+    topic_config_sensor = "homeassistant/sensor/splitflap_sensor_" + mdns + "/config";
+
+    mqttClient.setServer(mqttServer.c_str(), mqttPort);
+    mqttClient.setCallback([this](char* topic, byte* payload, unsigned int length) {
+        String message;
+        for (unsigned int i = 0; i < length; i++) {
+            message += (char)payload[i];
+        }
+        Serial.printf("[MQTT] Message received: %s\n", message.c_str());
+        if (display) {
+            float maxVel = settings.getFloat("maxVel");
+            display->writeString(message, maxVel, false);
+        }
+    });
+
+    connectToMqtt();
+}
+
+void SplitFlapMqtt::connectToMqtt() {
+    if (!mqttClient.connected()) {
+        Serial.println("[MQTT] Attempting to connect...");
+        String clientId = "SplitFlap-" + settings.getString("mdns");
+        if (mqttUser.length() > 0) {
+            mqttClient.connect(clientId.c_str(), mqttUser.c_str(), mqttPass.c_str());
+        } else {
+            mqttClient.connect(clientId.c_str());
+        }
+
+        if (mqttClient.connected()) {
+            Serial.println("[MQTT] Connected to broker");
+
+            String mdns = settings.getString("mdns");
+
+            String payload_text = "{"
+                "\"name\":\"" + settings.getString("name") + "\","
+                "\"unique_id\":\"splitflap_text_" + mdns + "\","
+                "\"command_topic\":\"" + topic_command + "\","
+                "\"availability_topic\":\"" + topic_avail + "\","
+                "\"device\":{"
+                    "\"identifiers\":[\"splitflap_" + mdns + "\"],"
+                    "\"name\":\"" + settings.getString("name") + "\","
+                    "\"manufacturer\":\"SplitFlap\","
+                    "\"model\":\"SplitFlap Display\","
+                    "\"sw_version\":\"1.0.0\""
+                "}"
+            "}";
+
+            String payload_sensor = "{"
+                "\"name\":\"" + settings.getString("name") + " (Sensor)\","
+                "\"unique_id\":\"splitflap_sensor_" + mdns + "\","
+                "\"state_topic\":\"" + topic_state + "\","
+                "\"availability_topic\":\"" + topic_avail + "\","
+                "\"device_class\":\"none\","
+                "\"entity_category\":\"diagnostic\","
+                "\"device\":{"
+                    "\"identifiers\":[\"splitflap_" + mdns + "\"],"
+                    "\"name\":\"" + settings.getString("name") + "\","
+                    "\"manufacturer\":\"SplitFlap\","
+                    "\"model\":\"SplitFlap Display\","
+                    "\"sw_version\":\"1.0.0\""
+                "}"
+            "}";
```

Comment:
I started using `JsonDocument` instead of string concatenation which makes building json payloads a bit easier to manage. Probably not quite as efficient though.